node-lib/src/config_files/p2p.rs

```diff
@@ -66,6 +66,13 @@ pub struct P2pConfigFile {
     pub disable_noise: Option<bool>,
     /// Optional list of boot node addresses to connect.
     pub boot_nodes: Option<Vec<IpOrSocketAddress>>,
+    /// If true, the node will assume that the nodes specified in `boot_nodes` won't be able
+    /// to provide it with fresh blocks.
+    /// This is used by p2p-test, where `boot_nodes` are other test nodes
+    /// that have no knowledge about nodes on the public network. Setting this to
+    /// true will force the node to make a dns seed query early if its peerdb only contains
+    /// addresses specified via `boot_nodes`.
+    pub boot_nodes_will_stall: Option<bool>,
```

Comment:
Well, `force_dns_seed_query` doesn't reflect what will happen exactly, because the query is not unconditional - it's performed only if peerdb has no addresses other than those specified in `boot_nodes`.